zoobc/zoobc-core
# fix require-escrow dataset attribute behavior

## Description
Fix condition in the transaction validation for accounts having a dataset set to require escrow `AccountDatasetEscrowApproval`.

## Breakdown
- [x] Fix the condition checking of the escrow attibute of a transaction
- [x] Add additional condition to only consider if that attribute is set by the same account having it, and a condition to only allow the escrow transactions to these accounts are having himself as approval

## Reference Issue
Close #1421

## Step to Test (optional)
1. set an account dataset to an account `AccountDatasetEscrowApproval` with any value
2. send a transaction having the account as recipient
3. observe that only satisfying transaction gets through
